### PR TITLE
feat: sidebar updates

### LINKED ISSUE #372: Add 'collected by' to profile sidebar + general profile sidebar updates
# **Collected by - [Design](https://www.figma.com/file/DC1T92jSoSIw9rvvEJytNK/?node-id=6024%3A12079)**

- [x] If a wallet has created NFTs that have sold to (or are owned by) other wallets, show a 'collected by' preview that is clickable, which opens a modal with the list of collector wallets. Use the same approach as 'followed by'.

- [x] If the wallet has created NFTs but they don't have owners (collectors), don't show the 'collected by' preview in the profile details. 

Screen Shot 2022-04-22 at 6 13 08 pm
Screen Shot 2022-04-22 at 6 13 18 pm
Screen Shot 2022-04-22 at 6 13 28 pm

### PR DESCRIPTION
**Pull request by milorue**
fixes #372 
- added collected by section of sidebar
- general cleanup of sidebar
- button placements
- No item CTAs for collected and created
image

### CODE REVIEW

File: src/common/components/home/FeaturedProfilesSection.tsx [hunk 1]
```diff
@@ -228,7 +228,7 @@ const PreviewContainer: FC<any> = (props) => {
   );
 };
 
-const FollowUnfollowButtonDataWrapper: VFC<{ targetPubkey: string; className?: string }> = ({
+export const FollowUnfollowButtonDataWrapper: VFC<{ targetPubkey: string; className?: string }> = ({
```
Comment (austinmilt) [inline src/common/components/home/FeaturedProfilesSection.tsx:231]: If you're gonna use this outside of `FeaturedProfilesSection`, it should go into `FollowUnfollowButton`

**Review by austinmilt — CHANGES_REQUESTED** [review]
mostly minor stuff, but I think `FollowUnfollowButtonDataWrapper` should be moved if you want to use it outside of `FeaturedProfilesSection`

File: src/common/components/elements/ProfileContainer.tsx [hunk 2]
```diff
@@ -38,24 +39,31 @@ export const ProfileContainer: FC = ({ children }) => {
...
-                !twitterHandle && (
-                  <ConnectTwitterButton
-                    address={new PublicKey(profileData.publicKey)}
-                    connection={connection}
-                    wallet={anchorWallet}
-                    cluster={'mainnet-beta'}
-                    variant={'secondary'}
-                    style={{ background: 'rgb(33,33,33)', height: '37px', borderRadius: '18px' }}
+              <div className="mt-16 flex justify-center lg:justify-start">
+                {anchorWallet?.publicKey.toString() == profileData.publicKey.toString() &&
```
[the remark this answers is not in the document]
Comment (milorue) [reply src/common/components/elements/ProfileContainer.tsx:43]: resolved

File: src/common/components/elements/NoProfileItems.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+          </span>
+        </div>
+      );
+    case 'created':
+      return (
+        <div
+          className={`mt-12 flex flex-col justify-center gap-4 rounded-lg border border-gray-800 p-4 text-center`}
+        >
+          <span className="text-center text-2xl font-semibold">
+            {`You haven't created anything yet`}
```
[the remark this answers is not in the document]
Comment (milorue) [reply src/common/components/elements/NoProfileItems.tsx:39]: i need to escape the quote

File: src/common/components/elements/NoProfileItems.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+const NoProfileItems: FC<NoProfileItemsProps> = ({ variant = 'collected' }) => {
+  switch (variant) {
+    case 'collected':
+      return (
+        <div
+          className={`mt-12 flex w-full flex-col rounded-lg border border-gray-800 p-4 text-center`}
+        >
+          <span className="text-center text-2xl font-semibold">
+            {`You haven't collected anything yet`}
```
[the remark this answers is not in the document]
Comment (milorue) [reply src/common/components/elements/NoProfileItems.tsx:19]: same as below

File: src/common/components/elements/FollowerCount.tsx [hunk 5]
```diff
@@ -124,20 +147,68 @@ const FollowedBy: FC<FollowedByProps> = ({ onOtherFollowersClick }) => {
...
+  onOtherCollectedClick?: VoidFunction;
+};
+
+const CollectedBy: FC<CollectedByProps> = ({ onOtherCollectedClick }) => {
+  const { publicKey } = useProfileData();
+  const { data, loading } = useGetCollectedByQuery({
+    variables: { creator: publicKey },
+  });
+  if (loading) return null;
+  let collectedProfiles: TwitterProfile[] = [];
```
[the remark this answers is not in the document]
Comment (milorue) [reply src/common/components/elements/FollowerCount.tsx:160]: resolved

File: src/common/components/home/FeaturedProfilesSection.tsx (hunk 1 above)
Comment (milorue) [reply src/common/components/home/FeaturedProfilesSection.tsx:231]: Agreed

**Pull request merged**